torch/_dynamo/output_graph.py

```diff
@@ -73,6 +73,21 @@ class OutputGraphState(NamedTuple):
     nn_modules: Optional[Dict[str, torch.nn.Module]]
     side_effects: SideEffects
     timestamp: int
+    name_to_input: OrderedDict[str, Optional[fx.Proxy]]
+
+    def diff(self, other: "OutputGraphState", *, prefix: str = "") -> Optional[str]:
+        for k in self._fields:
+            if k == "side_effects":
+                r = self.side_effects.diff(other.side_effects)
+                if r is not None:
+                    return r
+                continue
+
+            sv = getattr(self, k)
+            ov = getattr(other, k)
```

Comment:
what if other is a superset of self? Every key in self would match, but other would have more keys, so there is a diff, but we missed it. Do you need to support this case?

Reply:
It can't; fields is not a per-object property, it's a property of the named tuple itself.